Auth feature 2

snowzhop opened the pull request:
### JWT authentication

- added new classes for _JWT authentication_: **JwtAuthenticationFilter** and **JwtAuthorizationFilter**;
- added new class which includes JWT properties;
- **SecurityConfiguration** extended for JWT auth;
- added new method for user registration;
- user class field 'login' renamed to 'username'.

snowzhop committed:
Added configuration classes for JWT-auth

snowzhop committed:
Added new class for authentication

snowzhop committed:
Added new dependecies: spring-security, java-jwt

snowzhop committed:
Added new classes for using 'User DB'

snowzhop committed:
'Login' changed to 'Username'

snowzhop committed:
Added 'findByUsername' method

snowzhop committed:
'Login' changed to 'Username'

snowzhop committed:
Simple formatting changes

snowzhop committed:
Deleted class for early debugging

snowzhop committed:
Added simple user registration

IReallyHateLogins commented on Spring-server/src/main/java/ikpi63holding/postclient/JwtProperties.java:
```diff
@@ -0,0 +1,8 @@
+package ikpi63holding.postclient;
+
+public class JwtProperties {
+    public static final String SECRET = "ikpi63HOLDING";
```
So secret, it's in our open repository. I think we should put most of this class contents in app proprties so it can be changed without compiling.

ValentinBodrov replied:
Agreed

snowzhop replied:
Yes, we should hide it

IReallyHateLogins commented on Spring-server/src/main/java/ikpi63holding/postclient/JwtAuthorizationFilter.java:
```diff
@@ -0,0 +1,66 @@
+
+    private Authentication getUsernamePasswordAuthentication(HttpServletRequest request) {
+        String token = request.getHeader(JwtProperties.HEADER_STRING)
+                .replace(JwtProperties.TOKEN_PREFIX,"");
```
Why do we replace headers?

snowzhop replied:
Because JWT token looks like "Bearer <payload>". This replacing method deletes useless "Bearer " in header for using token in the future

IReallyHateLogins commented on Spring-server/src/main/java/ikpi63holding/postclient/ApiController.java:
```diff
@@ -122,4 +124,27 @@ void deleteUser(@PathVariable Long id) {
+
+        return repository.save(new User(
+            newUser.getUsername(),
+            passwordEncoder.encode(newUser.getPassword()),
```
What happends when password is null? Maybe we should check for it implicitly.

snowzhop replied:
If password is null we'll get NullPointerException but i think we should check it before sending of the registration request

IReallyHateLogins reviewed (dismissed):
Seems great, I will say more after some testing.

Daniel1898 reviewed (dismissed):
Good job!

ValentinBodrov commented on Spring-server/src/main/java/ikpi63holding/postclient/JwtAuthorizationFilter.java:
```diff
@@ -0,0 +1,66 @@
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+// import org.springframework.security.web.AuthenticationEntryPoint;
```
Take these comments off in future comments please.

ValentinBodrov commented on Spring-server/src/main/java/ikpi63holding/postclient/LoginViewModel.java:
```diff
@@ -0,0 +1,22 @@
+    public void setPassword(String password) {
+        this.password = password;
+    }
+}
```
Is this file necessary? We have already **User.java**, you could use some fields of them by auth process

snowzhop replied:
Oh, i forgot about User class.

ValentinBodrov reviewed (commented):
I didn't understand some points below

snowzhop committed:
Deleted and replaced LoginViewModel

snowzhop committed:
Deleted useless ExceptionHandler

IReallyHateLogins committed:
Moved undertow config to spring app entry config

IReallyHateLogins committed:
Moved jwt properties to app properties

Binded JwtProperties to jwt in app properties, added JwtProperties to DI, also added more DI to SecurityConfiguration, now passing properties to filters through constructor because they are not  managed by DI.
Also done some reformatting.

IReallyHateLogins committed:
Merge pull request #24 from Valbod-I-am-a-Swiftie/auth-feature2-config

Auth feature2 config

ValentinBodrov reviewed (approved):
It's allright, there's no things those call any questions
